Examples/Scripts/Python/itk_from_geomodel_gen1.py

```diff
@@ -0,0 +1,442 @@
+import random
+import re
+
+import acts
+import argparse
+from acts import (
+    logging,
+    GeometryContext,
+    CylindricalContainerBuilder,
+    DetectorBuilder,
+    GeometryIdGenerator,
+)
+from acts import geomodel as gm
+from acts import examples
+
+from propagation import runPropagation
+
+
+def merge(a, b):
+    if isinstance(a, list) and isinstance(b, list):
+        return a + b
+
+    assert isinstance(a, Hierarchy) and isinstance(b, Hierarchy)
+    assert a.level == b.level
+    assert a.max_level == b.max_level
+    assert a.settings == b.settings
+
+    merged = Hierarchy(a.level, a.max_level, a.settings)
+
+    keys = set([*a.keys(), *b.keys()])
+
+    for key in keys:
+        if key in a and not key in b:
+            merged[key] = a[key]
+        elif key in b and not key in a:
+            merged[key] = b[key]
+        else:
+            merged[key] = merge(a[key], b[key])
+
+    return merged
+
+
+class HierarchySetting:
+    """
+    This class will be passed to all child hierarchies by reference (compared to a single bool, that will be passed by value)
+    This way, we can manipulate all child trees with one call
+    """
+
+    allow_insertion = True
+
+
+class Hierarchy(dict):
+    """
+    Small utility class, that allows to easily construct fixed depth hierarchies
+    After maxlevel is reached, a list is inserted instead of another Hierarchy
+    """
+
+    def __init__(self, level, max_level, settings=HierarchySetting()):
+        self.level = level
+        self.max_level = max_level
+        self.settings = settings
+
+    def __getitem__(self, key):
+        if key is None:
+            assert self.level < self.max_level
+            broadcast = Hierarchy(self.level + 1, self.max_level, self.settings)
+            for k in self.keys():
+                broadcast = merge(broadcast, self[k])
+
+            return broadcast
+
+        if isinstance(key, list):
+            assert self.level < self.max_level
+            broadcast = Hierarchy(self.level + 1, self.max_level, self.settings)
+            for k in key:
+                broadcast = merge(broadcast, self[k])
+
+            return broadcast
+
+        if self.settings.allow_insertion and not key in self:
+            if self.level < self.max_level:
+                self[key] = Hierarchy(self.level + 1, self.max_level, self.settings)
+            else:
+                self[key] = []
+
+        return super().__getitem__(key)
+
+    def flatten(self):
+        """
+        Flatten out the hierarchy at a given level
+        """
+        l = []
+        if self.level < self.max_level:
+            for key in self.keys():
+                l += self[key].flatten()
+        else:
+            for key in self.keys():
+                l += self[key]
+
+        return l
+
+
+class ItkBuilder:
```

Comment:
Can't most (all?) of this be done in C++? I'm not sure it's necessarily a good idea to put so much logic and one-off infrastructure into python like this. If you write this in C++, you can just bind the outer interface to python and achieve the same thing with fewer downsides.